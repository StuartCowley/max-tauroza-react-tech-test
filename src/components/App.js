import React, { useState } from 'react';
import Search from './Search';
import '../styles/App.css';

const App = () => {
  const [searchResults, setSearchResults] = useState([]);

  return (
    <div className="App">
      <img
        className="logo"
        src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
        alt="Nasa logo"
      />
      <h1 className="main-title">Space Search</h1>
      <Search setSearchResults={setSearchResults} />
    </div>
  );
};

export default App;
