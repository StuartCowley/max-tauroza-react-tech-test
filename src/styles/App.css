.App {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.logo {
  margin-top: 20px;
  height: 108px;
  width: 192px;
  border-radius: 30px;
}

.main-title {
  margin-top: 30px;
  font-size: 80px;
  font-family: 'Audiowide';
  color: white;
}

@media only screen and (max-width: 615px) {
  .main-title {
    font-size: 10vw;
  }

  .logo {
    width: 30vw;
    height: auto;
  }
}
