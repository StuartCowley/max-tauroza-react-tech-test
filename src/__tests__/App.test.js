import { render, screen } from '@testing-library/react';
import App from '../components/App';

describe('App', () => {
  it('renders app component correctly', () => {
    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders an image', () => {
    render(<App />);
    expect(screen.getByAltText('Nasa logo')).toBeInTheDocument();
  });
});
