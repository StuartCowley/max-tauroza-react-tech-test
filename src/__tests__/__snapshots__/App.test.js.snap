// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders app component correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <img
      alt="Nasa logo"
      class="logo"
      src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
    />
    <h1
      class="main-title"
    >
      Space Search
    </h1>
    <form>
      <input
        class="searchBar"
        placeholder="Search"
        type="text"
      />
      <button
        class="searchButton"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
